pulp_rpm/app/access_policy.py

```diff
@@ -0,0 +1,54 @@
+from logging import getLogger
+
+from pulpcore.plugin.models import RepositoryVersion
+from pulpcore.plugin.access_policy import AccessPolicyFromDB
+
+from pulp_rpm.app.models.repository import RpmRepository
+
+_logger = getLogger(__name__)
+
+
+class CopyAccessPolicy(AccessPolicyFromDB):
+    """
+    Access policy for ContainerNamespaceViewSet.
+    """
+
+    def has_perms_to_copy(self, request, view, action):
+        """
+        Check if the source and destination repository matches the usernames permissions.
+
+        `Fail` the check at first missing permission.
+        """
+        serializer = view.serializer_class(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+
+        all_user_permissions = request.user.get_all_permissions()
+
+        for copy_action in serializer.data["config"]:
+            # Get destination repo object by its UUID
+            dest_repo_id = copy_action["dest_repo"].split("/")[-2]
+            dest_repo = RpmRepository.objects.get(pk=dest_repo_id)
+
+            # Check if user has permissions to destination repository
+            if not (
+                request.user.has_perm("rpm.modify_content_rpmrepository", dest_repo)
+                or "rpm.modify_content_rpmrepository" in all_user_permissions
+            ):
+                return False
+
+            # Get source repo object by its UUID
+            source_repo_id = copy_action["source_repo_version"].split("/")[-4]
+            source_repo_ver_number = copy_action["source_repo_version"].split("/")[-2]
+            source_repo_version = RepositoryVersion.objects.get(
```

Comment:
It looks like names of the parameters in these comments are out of date: https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/tasks/copy.py#L121-L122=

But I recommend not re-writing this code, we already have a way to do it here: https://github.com/pulp/pulp_rpm/blob/1bb89b0aff6c958c6b4ef2fd0b23548c6fe98a42/pulp_rpm/app/viewsets.py#L366-L369=

Start with copying the implementation and then see if you can figure out if it would be possible (and not messy) to share the implementation somehow.

Reply:
@pavelpicka Will continue the review after this is looked into

Reply:
@dralley rbac is repo centric so those two lines should be enough